File: lesson-02/BaseExample/Java/Consumer/src/main/java/ru/example/kafka/Consumer.java
package ru.example.kafka;

import org.apache.kafka.clients.consumer.ConsumerConfig;
import org.apache.kafka.clients.consumer.ConsumerRecord;
import org.apache.kafka.clients.consumer.ConsumerRecords;
import org.apache.kafka.clients.consumer.KafkaConsumer;
import org.apache.kafka.common.errors.WakeupException;
import org.apache.kafka.common.serialization.IntegerDeserializer;
import org.apache.kafka.common.serialization.StringDeserializer;
import java.time.Duration;
import java.util.Collections;
import java.util.Properties;

public class Consumer {
    public static void main(String[] args) {
        final String BOOTSTRAP_SERVERS = "localhost:9092";
        final String topic = "test";
        final String GROUP_ID = "g1";
        final String OFFSET_RESET = "earliest";

        Properties props = new Properties();
        props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, BOOTSTRAP_SERVERS);
        props.put(ConsumerConfig.GROUP_ID_CONFIG, GROUP_ID);
        props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, OFFSET_RESET);
        props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, IntegerDeserializer.class.getName());
        props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
        KafkaConsumer<Integer, String> consumer = new KafkaConsumer<>(props);

        try {
            consumer.subscribe(Collections.singleton(topic));
            while (true) {
                ConsumerRecords<Integer, String> records = consumer.poll(Duration.ofSeconds(1));
                for (ConsumerRecord<Integer, String> data : records) {
                    String message = String.format("%d\t%d\t%d\t%s", data.partition(), data.offset(), data.key(), data.value());
                    System.out.println(message);
                }
            }
        } catch (WakeupException e) {
            System.out.println(e.getLocalizedMessage());
        } finally {
            consumer.close();
        }
    }
}

File: lesson-02/BaseExample/Clojure/consumer/src/ru/vzaigrin/examples/kafka/base/clj/consumer.clj
(ns ru.vzaigrin.examples.kafka.base.clj.consumer
  (:gen-class)
  (:import
   (java.time Duration)
   (java.util Properties)
   (org.apache.kafka.clients.consumer ConsumerConfig KafkaConsumer)))

(defn -main
  "Base Kafka Consumer"
  []
  (let [topic "test"
        props (doto (Properties.)
                (.put ConsumerConfig/BOOTSTRAP_SERVERS_CONFIG "localhost:29092")
                (.put ConsumerConfig/KEY_DESERIALIZER_CLASS_CONFIG "org.apache.kafka.common.serialization.LongDeserializer")
                (.put ConsumerConfig/VALUE_DESERIALIZER_CLASS_CONFIG "org.apache.kafka.common.serialization.StringDeserializer")
                (.put ConsumerConfig/GROUP_ID_CONFIG "g1")
                (.put ConsumerConfig/AUTO_OFFSET_RESET_CONFIG "earliest")
                (.put ConsumerConfig/ENABLE_AUTO_COMMIT_CONFIG "false"))]
    (with-open [consumer (KafkaConsumer. props)]
      (.subscribe consumer [topic])
      (while true 
        (doseq [msg (.poll consumer (Duration/ofSeconds 1))]
          (printf "%d\t%d\t%d\t%s\n"
                  (.partition msg)
                  (.offset msg)
                  (.key msg)
                  (.value msg)))))))
